packages/core/src/FindContractVersion.ts

```diff
@@ -16,16 +26,13 @@ try {
  * @param {string} contractAddress address of the contract in question
  * @return {Object} contract name & version
  */
-async function findContractVersion(contractAddress, web3) {
-  assert(web3, "Web3 object must be provided");
-  assert(contractAddress, "Contract address must be provided");
-
+export async function findContractVersion(contractAddress: string, web3: Web3) {
   // Note: there is an unknown issue in web3.js that means that the `getCode` syntax does not function correctly in
   // production. However, ethers has proven to work correctly in production. The code below is a patch to still enable
   // this module to work while we find a better long term solution for the web3.js issue. If running within unit tests
   // then the web3.js version is required as it is scope according to the unit test.
   let contractCode;
-  if (global.web3) {
+  if (((global as unknown) as { web3: Web3 | undefined }).web3) {
```

Comment:
This is pretty ugly, but I guess the global.web3 object could be different types depending on the provider?

Reply:
Yeah, the issue is that typescript doesn't know what properties global has, so we have to basically cast it before testing it. And it forces you to cast to unknown before the custom type :/.